src/Storages/MergeTree/SimpleMergeSelector.cpp

```diff
@@ -152,6 +153,9 @@ void selectWithinPartition(
         if (begin > 1000)
             break;
 
+        if (!parts[begin].can_participate_in_merges)
```

Comment:
Why we check this here? SimpleMergeSelector already quite complicated. I think we can put this check in the `selectPartsToMerge` to start a new parts interval. Logic in selectors shouldn't change.

Reply:
We check it here because otherwise selectors become even more complicated because they don't support gaps in parts - parts could be merged only in contiguous ranges. So I have two options here: 1) make some logic which would give few contiguous ranges to selector and after that implement some logic which shall select from that list of ranges - that is another merge selector actually; 2) select range of subset of parts and then possibly cancel that selection - which may block all merges if any part is on non-mergeable volume.